torch/_inductor/dependencies.py

```diff
@@ -239,6 +241,17 @@ def index_expr(self, index: sympy.Expr, dtype) -> str:
         self._index_exprs.add(IndexExprDep(*self.canonicalize(index)))
         return f"index_expr({sympy_str(index)}, {dtype})"
 
+    def bucketize(
+        self,
+        values,
+        offsets_name: str,
+        offsets_size: sympy.Expr,
+        indexing_dtype: torch.dtype,
+        right: bool,
+    ):
+        self._reads.add(StarDep(offsets_name))
```

Comment:
Why not :

`self._writes.add(MemoryDep(offsets_name, *self.canonicalize(offsets_size)))` ?

Reply:
AFAIK, the `index` field is intended to refer to the indices that are accessed during the read. In this case, the binary search needs the raw `offsets_ptr` and then it will binary search over the offsets, so it actually does need the entire tensor and we don't actually know the indices it will need, because that's data-dependent.

offsets_size also isn't the index we're accessing, it's the full size of the tensor.

but I could be wrong about this interpretation, if so LMK

Reply:
Should we just add the dependency as the entire tensor then ?